plugins/debuginfo-install.py

```diff
@@ -74,59 +75,125 @@ def configure(self):
-            logger.info(_("No debuginfo packages available to install"))
+            # Remove debuginfo packages if their base packages are in the query.
+            # They can get there through globs and they break the installation
+            # of debug packages with the same version as the installed base
+            # packages. If the base package of a debuginfo package is not in
+            # the query, the user specified a debug package on the command
+            # line. We don't want to ignore those, so we will install them.
+            # But, in this case the version will not be matched to the
+            # installed version of the base package, as that would require
+            # another query and is further complicated if the user specifies a
+            # version themselves etc.
+            for name in list(package_dict.keys()):
+                if name.endswith(Package.DEBUGINFO_SUFFIX):
+                    if name[:-debuginfo_suffix_len] in package_dict:
+                        package_dict.pop(name)
+                if name.endswith(Package.DEBUGSOURCE_SUFFIX):
+                    if name[:-debugsource_suffix_len] in package_dict:
+                        package_dict.pop(name)
+
+            # attempt to install debuginfo and debugsource for the highest
+            # listed version of the package (in case the package is installed,
+            # only the installed version is listed)
+            for packages in package_dict.values():
+                package = packages[0]
+
+                if packages[0]._from_system:
+                    # for packages from system (installed) there can be more
+                    # packages with different architectures listed and we want
+                    # to install debuginfo for all of them
+                    for package in packages:
+                        if not self._install_debug_from_system(package, package.debug_name):
+                            self.debuginfo_missing.add(str(package))
+
+                        if not self._install_debug_from_system(package, package.debugsource_name):
+                            self.debugsource_missing.add(str(package))
+
+                else:
+                    # if the first package is not from the system, we just want its name
+                    package = packages[0]
+                    if package.name.endswith(Package.DEBUGINFO_SUFFIX) \
+                            or package.name.endswith(Package.DEBUGSOURCE_SUFFIX):
+
+                        self._install((package, ))
+                        continue
+
+                    if solution["nevra"] is None:
+                        print("TODO We have this package and it may be the wrong one: %s", package)
+
+                    if not self._install_debug(package.debug_name, solution["nevra"]):
+                        self.debuginfo_missing.add(str(package)) # TODO check this is the correct package being logged - could be different architecture etc.
```

Comment:
To report latest packages for all arch it can be resolve by:
```
    def _latest_arch_representant(self, packages):
        """Packages should be sorted list"""
        latest_per_arch = [packages[0]]
        arch = packages[0].arch
        for pkg in packages[1:]:
            if arch != pkg.arch:
                latest_per_arch.append(pkg)
                arch = pkg.arch
        return latest_per_arch
```
and self.debuginfo_missing.add(str(package)) could be replaced by:
```
for pkg in _latest_arch_representant(packages):
    self.debuginfo_missing.add(str(pkg))
```

Reply:
I've split the _Could not find ..._ messages instead and for installed packages I print out the full NEVRA, for available packages I print out NEVR, omitting the architecture, of which there can be multiple and it can be confusing listing the same package multiple times with different architectures.

Reply:
ok